Prepare for 4.0.0

nirinchev opened the pull request:
🎉

nirinchev commented:
@bmunkholm feel free to review the changelog here: https://github.com/realm/realm-dotnet/blob/fe10a22a284302d439b9ebdcc4f3e48aa7ff224a/CHANGELOG.md

bmunkholm commented:
Isn't ROS 3.21 required? 
Could we add an internal section in the release note listing the sync version as we do for other SDKs now?
Haven't we fixed bugs through upgraded sync version?

nirinchev commented:
@bmunkholm I've updated the ROS requirement - it's actually 3.23.1 and added changelog entries for the fixes from Core/Sync - I didn't find any sync fixes that affect the SDKs directly, but I added the ones from Core that seemed relevant.

bmunkholm commented on CHANGELOG.md:
```diff
@@ -61,12 +61,21 @@
+* Improved query performance for unindexed string columns when the query has a long chain of OR conditions. (Core upgrade)
+* Improved performance of encryption and decryption significantly by utilizing hardware optimized encryption functions. (Core upgrade)
+* Compacting a realm into an encrypted file could take a really long time. The process is now optimized by adjusting the write buffer size relative to the used space in the realm. (Core upgrade)
+* The string-based query parser (`results.Filter("...")`) now supports readable timestamps with a 'T' separator in addition to the originally supported "@" separator. For example: `"startDate > 1981-11-01T23:59:59:1"`. (Core upgrade)
```
Forgive my ignorance - does .Net also support string based queries now?

nirinchev replied:
Yes - we added support in 3.0.0 as we didn't want to hold functionality hostage to our LINQ support: https://github.com/realm/realm-dotnet/blob/master/CHANGELOG.md#enhancements-5

bmunkholm replied:
Cool - Where does one find docs for that? (I did actually try to find it without success)

bmunkholm commented:
Do we have a PR or Issue for doc updates?

nirinchev commented:
Here's the PR for docs: https://github.com/realm/realm.io/pull/4641

nirinchev commented:
@bmunkholm the .net API docs have GitHub links for method implementations that unfortunately can't resolve tags (they only work with branches) so we keep the release branches around.
